src/game/Unit.cpp

```diff
@@ -339,7 +339,7 @@ bool Unit::UpdateMeleeAttackingState()
     if (!victim || IsNonMeleeSpellCasted(false))
         return false;
 
-    if (!isAttackReady(BASE_ATTACK) && !(isAttackReady(OFF_ATTACK) && haveOffhandWeapon()))
+    if (!isAttackReady(BASE_ATTACK) && (haveOffhandWeapon() && !(isAttackReady(OFF_ATTACK))))
```

Comment:
I think wrong change.  Check expected skip mo any weapon attack cases . In final form it true only if  unit have offhand weapon and so always false for unit without offhand weapon